### ISSUE TITLE
UndefinedPropertyAssignment on ad-hoc Eloquent attribute assignment ($model->foo = value)

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#779]
**Describe the bug**

Eloquent models route all property writes through `Model::__set()` / `setAttribute()`, so attaching an ad-hoc attribute is valid Laravel code — commonly used to decorate a model instance with a computed value before serialization. The plugin reports `UndefinedPropertyAssignment` for any attribute not declared via `@property`, which produces a large volume of false positives in real applications.

**Code to reproduce**

```php
<?php

use Illuminate\Database\Eloquent\Model;

class User extends Model {
    protected $fillable = ['email'];
}

$user = new User();
$user->md5 = md5(strtolower($user->email)); // UndefinedPropertyAssignment
```

Error:

```
UndefinedPropertyAssignment - Instance property Illuminate\Database\Eloquent\Collection::$md5 is not defined
```

(Also note the message names `Collection` instead of `User` — that is a secondary symptom of the `findOrFail` generic widening issue being filed separately.)

**Real-world occurrence**

pterodactyl/panel UserController.php:144 — https://github.com/pterodactyl/panel/blob/760c3db300b6ff51378edff4d0e74b3855f57ca0/app/Http/Controllers/Admin/UserController.php#L144

**Suggested behaviour**

Every subclass of `Illuminate\Database\Eloquent\Model` has `__set`, so `UndefinedPropertyAssignment` on direct property writes against a model instance is always a false positive unless the user has explicitly opted into stricter model typing. Either:

1. Suppress `UndefinedPropertyAssignment` on Eloquent models by default, or
2. Emit a separate, opt-in issue code (e.g. `UndeclaredEloquentAttributeWrite`) so that the existing dynamic-attribute convention doesn't drown out the plugin's `@property` inference work.

**Impacted Versions**

```
laravel/framework      11.51.0
psalm/plugin-laravel   3.8.4
vimeo/psalm            6.16.1
```

**Psalm config**

```xml
<psalm errorLevel="6" findUnusedBaselineEntry="true" findUnusedCode="false">
  <projectFiles>
    <directory name="app"/>
    <ignoreFiles><directory name="vendor"/></ignoreFiles>
  </projectFiles>
  <plugins>
    <pluginClass class="Psalm\LaravelPlugin\Plugin"/>
  </plugins>
</psalm>
```
